subprojects/packagefiles/protobuf-c/meson.build

```diff
@@ -33,13 +36,19 @@ protoc_c_src = [
   'protoc-c/main.cc',
 ]
 
-protoc_deps = dependency('protobuf',
-                         fallback: ['protobuf', 'protoc_dep'],
-                         version: '>= 3.0.0')
+protoc_deps = dependency(
+  'protobuf',
+  fallback: ['protobuf', 'protoc_dep'],
+  version: '>= 3.0.0'
+)
+
+protoc_c = executable(
+  'protoc-c',
+  protoc_c_src,
+  cpp_args: '-DPACKAGE_STRING="protobuf-c"',
+  include_directories: include_directories('.'),
+  dependencies: protoc_deps
+)
 
-protoc_c = executable('protoc-c',
-                      protoc_c_src,
-                      cpp_args: '-DPACKAGE_STRING="protobuf-c"',
-                      include_directories: include_directories('protoc-c',
-                                                               'protobuf-c'),
-                      dependencies: protoc_deps)
+run_command('protoc', '--cpp_out', meson.current_build_dir(), '--proto_path', meson.current_source_dir(), meson.current_source_dir() + '/protobuf-c/protobuf-c.proto', check : true)
+run_command(protoc_c, meson.current_build_dir() + '/protobuf-c/protobuf-c.pb.cc', meson.current_build_dir() + '/protobuf-c/protobuf-c.pb.h')
```

Comment:
What is this doing??? You cannot use a build target in run_command anyway, but on general principle why would this be a run_command rather than a custom_target?

However, it is also not entirely clear to me what it is trying to accomplish. Maybe it is supposed to test running the built software? Wouldn't that be better as a `test()`?

Reply:
eh no. It's building files needed for compilation.

Reply:
It's not clear to me where/when this is needed for compilation, and it still needs to be a custom_target (or generator) in that case.

Reply:
sounds about right.